### PR TITLE
Add medication service method to create a patient medication

### PR DESCRIPTION
**Pull request by gffking**
Also add service for discontinuing medications.

### CODE REVIEW

**Review by fterrag — APPROVED** [review]

File: descontinued_medication.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+type DiscontinuedMedicationService struct {
+	client *Client
+}
+
+type DiscontinuedMedicationCreate struct {
+	MedOrder             int64  `json:"med_order"`
+	DiscontinueDate      string `json:"discontinue_date,omitempty"`
+	Reason               string `json:"reason,omitempty"`
+	IsDocumented         bool   `json:"is_documented"`
```
Comment (elliehastings) [inline descontinued_medication.go:32]: It looks like this one supports a null, should it also be `omitempty`?

File: descontinued_medication.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	Strength       string   `json:"strength"`
+	Form           string   `json:"form"`
+}
+
+type FindDiscontinuedMedicationsOptions struct {
+	*Pagination
+
+	Patient         []int64   `url:"patient,omitempty"`
+	Practice        []int64   `url:"practice,omitempty"`
+	DocumentDateLT  time.Time `url:"document_date__lt,omitempty"`
```
Comment (elliehastings) [inline descontinued_medication.go:93]: The API lists `lte`/`gte` twice but not `lt`/`gt`, I assume that's a docs error? 😅

**Review by elliehastings — APPROVED** [review]

File: descontinued_medication.go (hunk 2 above)
Comment (gffking) [reply descontinued_medication.go:93]: Nope, docs are right, but I missed that the first part is different for two of them (`document_date` vs `discontinue_date`).

File: descontinued_medication.go (hunk 1 above)
Comment (gffking) [reply descontinued_medication.go:32]: Good catch, will change.

**Pull request merged**